packages/react-components/lib/tasks/task-timeline.tsx

```diff
@@ -48,47 +47,99 @@ const StyledTimeLine = styled((props: TimeLinePropsWithRef) => <Timeline {...pro
   }),
 );
 
+function nestedEvents(eventsStates: { [key: string]: EventState } | undefined, child: number) {
+  if (eventsStates !== undefined) {
+    const deps = eventsStates[child] ? eventsStates[child].deps : [];
+    return (
+      <TreeItem
+        key={`event-${child}`}
+        nodeId={`parent-${eventsStates[child].name}`}
+        label={eventsStates[child].name}
+      >
+        {deps?.map((id) => {
+          if (eventsStates[id].deps?.length) {
+            nestedEvents(eventsStates, id);
+          } else {
+            return (
+              <TreeItem
+                key={eventsStates[child].name}
+                nodeId={`child-${id}`}
+                label={eventsStates[id].name}
+              ></TreeItem>
+            );
+          }
+        })}
+      </TreeItem>
+    );
+  }
+}
+
+function colorDot(eventsStates: { [key: string]: EventState } | undefined) {
+  let standby: Boolean = true;
+  for (const id in eventsStates) {
+    const event: EventState = eventsStates[id];
```

Comment:
Since you are not using `id` in any other meaningful way, prefer `for...of` instead.

```ts
  for (const event of eventsStates) {
```

Reply:
I tried this but kept giving me a compilation error somehow